pkg/execution/ratelimit/rueidis.go

```diff
@@ -0,0 +1,158 @@
+package ratelimit
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/inngest/inngest/pkg/inngest"
+	"github.com/rueian/rueidis"
+)
+
+const (
+	redisCASMissingKey = "key does not exist"
+	redisCASScript     = `
+local v = redis.call('get', KEYS[1])
+if v == false then
+  return redis.error_reply("key does not exist")
+end
+if v ~= ARGV[1] then
+  return 0
+end
+redis.call('setex', KEYS[1], ARGV[3], ARGV[2])
+return 1
+`
```

Comment:
nit:
make it an actual lua script as a follow up.

Reply:
lol ya, i was being particularly lazy for this small script;  didnt want to have to deal with the tediousness of `fs.Embed` given there's such a small single script.